File: app/src/main/java/com/example/soroushprofile/models/ChannelConversation.java
package com.example.soroushprofile.models;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.annimon.stream.Optional;

import java.util.List;

public class ChannelConversation extends ConversationThread {

    private String name;
    private Integer subscriber;
    private Optional<List<String>> media;
    private Optional<String> description;

    ChannelConversation(Optional<List<String>> media, Optional<String> description, String name, Integer subscriber) {
        this.name = name;
        this.subscriber = subscriber;
        this.media = media;
        this.description = description;
    }

    @NonNull
    @Override
    public String getTitle() {
        return name;
    }

    @Nullable
    @Override
    public Integer getAvatar() {
        return null;
    }

    @NonNull
    @Override
    public ConversationType getType() {
        return ConversationType.channel;
    }

    @Override
    public Optional<List<String>> getMedia() {
        return media;
    }

    @Override
    public Optional<String> getDescription() {
        return description;
    }

    public Integer getSubscriber() {
        return subscriber;
    }
}

File: app/src/main/java/com/example/soroushprofile/models/ChannelConversation.kt
package com.example.soroushprofile.models

class ChannelConversation internal constructor(override val media: List<String>?,
                                               override val description: String?,
                                               override val title: String,
                                               val subscriber: Int?) : ConversationThread() {

    override val avatar: Int?
        get() = null

    override val type: ConversationType
        get() = ConversationType.Channel
}
